tests/utils.py

```diff
@@ -1453,3 +1453,42 @@ class TypeMap(NamedTuple):
     TypeMap("geography", "geography", GeographyType()),
     TypeMap("geometry", "geometry", GeometryType()),
 ]
+
+
+def attr_to_dict(span):
+    dict_attr = {}
+    for name in span.attributes:
+        dict_attr[name] = span.attributes[name]
+    dict_attr["code.filepath"] = os.path.basename(dict_attr["code.filepath"])
+    dict_attr["status_code"] = span.status.status_code
+    dict_attr["status_description"] = span.status.description
+    return dict_attr
+
+
+def span_extractor(dict_exporter):
+    spans = []
+    raw_spans = dict_exporter.get_finished_spans()
+    for span in raw_spans:
+        spans.append((span.name, attr_to_dict(span), span))
+    return spans
+
+
+def check_single_answer(result, expected_answer):
```

Comment:
similar, check_single_answer -> check_single_span_result etc

Reply:
if nobody else is calling check_single_answer, there is probably no need to have a separate function for this

Reply:
I will add docstrings and type hints.
This is a helper function for check_answer, it is not meant to directly be used in tests, so I have to keep this